components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt

```diff
@@ -510,19 +534,19 @@ open class Nimbus(
     }
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal fun recordExposure(experimentId: String) {
+    internal fun recordExposure(featureId: String) {
         dbScope.launch {
-            recordExposureOnThisThread(experimentId)
+            recordExposureOnThisThread(featureId)
         }
     }
 
     // The exposure event should be recorded when the expected treatment (or no-treatment, such as
     // for a "control" branch) is applied or shown to the user.
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     @WorkerThread
-    internal fun recordExposureOnThisThread(experimentId: String) = withCatchAll {
+    internal fun recordExposureOnThisThread(featureId: String) = withCatchAll {
         val activeExperiments = getActiveExperiments()
-        activeExperiments.find { it.slug == experimentId }?.also { experiment ->
+        activeExperiments.find { it.featureIds.contains(featureId) }?.also { experiment ->
```

Comment:
This means:
1. `getExperimentBranch` by experiment slug is now not recording exposure events.
2. `getExperimentBranch` by feature id did not record exposure events prior to this change?

Reply:
Correct. My thinking was that get branch by experiment slug was deprecated and this would help to support that. yes, we may have been missing exposure events for a couple of experiments since it was trying to match feature ids against the slug.